aspnetcore/mvc/controllers/filters.md

```diff
@@ -178,97 +222,123 @@ In the following code, both the `ShortCircuitingResourceFilter` and the `AddHead
+* `TypeFilterAttribute` can optionally accept constructor arguments for the type.
 
-* Doesn't require any arguments.
-* Has constructor dependencies that need to be filled by DI.
+When using `TypeFilterAttribute`, setting `IsReusable` is a hint that the filter instance *may* be reused outside of the request scope it was created within. The framework provides no guarantees that a single instance of the filter will be created. `IsReusable` should not be used when with a filter that depends on services with a lifetime other than singleton.
 
-You can use your own named attribute on classes and methods instead of `[TypeFilter(typeof(FilterType))]`). The following filter shows how this can be implemented:
+The following example shows how to pass arguments to a type using `TypeFilterAttribute`:
 
-[!code-csharp[](./filters/sample/src/FiltersSample/Filters/SampleActionFilterAttribute.cs?name=snippet_TypeFilterAttribute&highlight=1,3,7)]
+[!code-csharp[](../../mvc/controllers/filters/sample/FiltersSample/Controllers/HomeController.cs?name=snippet_TypeFilter&highlight=1,2)]
+[!code-csharp[](../../mvc/controllers/filters/sample/FiltersSample/Filters/LogConstantFilter.cs?name=snippet_TypeFilter_Implementation&highlight=6)]
 
-This filter can be applied to classes or methods using the `[SampleActionFilter]` syntax, instead of having to use `[TypeFilter]` or `[ServiceFilter]`.
+<!-- 
+https://localhost:44358/home/hi?name=joe
+VS debug window shows 
+FiltersSample.Filters.LogConstantFilter:Information: Method 'Hi' called
+-->
 
 ## Authorization filters
 
 *Authorization filters*:
 
+* Are the first filters run in the filter pipeline.
 * Control access to action methods.
-* Are the first filters to be executed within the filter pipeline. 
-* Have a before method, but no after method. 
+* Have a before method, but no after method.
+
+Custom authorization filters require a custom authorization framework. Prefer configuring the authorization policies or writing a custom authorization policy over writing a custom filter. The built-in authorization filter:
 
-You should only write a custom authorization filter if you are writing your own authorization framework. Prefer configuring your authorization policies or writing a custom authorization policy over writing a custom filter. The built-in filter implementation is just responsible for calling the authorization system.
+* Calls the authorization system.
+* Does not authorize requests.
 
-You shouldn't throw exceptions within authorization filters, since nothing will handle the exception (exception filters won't handle them). Consider issuing a challenge when an exception occurs.
+Do **not** throw exceptions within authorization filters:
+
+* The exception will not be handled.
+* Exception filters will not handle the exception.
+
+Consider issuing a challenge when an exception occurs in an authorization filter.
 
 Learn more about [Authorization](xref:security/authorization/introduction).
 
 ## Resource filters
```

Comment:
To match most of the other sections, this should have the following as the first bit of content:

> *Resource filters*:

There are a couple of others that don't follow the pattern, but I won't call them out individually.